packages/lesswrong/server/updateScores.ts

```diff
@@ -166,12 +173,78 @@ export const batchUpdateScore = async ({collection, inactive = false, forceUpdat
         }
       }
     },
-  ])
+  ]);
+
+  return items.toArray();
+}
+
+const getPgCollectionProjections = (collectionName: CollectionNameString) => {
+  const proj = {
+    _id: '"_id"',
+    postedAt: '"postedAt"',
+    scoreDate: '"postedAt" AS "scoreDate"',
+    score: '"score"',
+    baseScore: '"baseScore"',
+  };
+  switch (collectionName) {
+    case "Posts":
+      proj.scoreDate = `(CASE WHEN "frontpageDate" IS NULL
+        THEN "postedAt"
+        ELSE "frontpageDate" END) AS "scoreDate"`;
+      proj.baseScore = `("baseScore" +
+        (CASE WHEN "frontpageDate" IS NULL THEN 0 ELSE 10 END) +
+        (CASE WHEN "curatedDate" IS NULL THEN 0 ELSE 10 END)) AS "baseScore"`;
+      break;
+    case "Comments":
+      const {base, magnitude, duration, exponent} = getSubforumScoreBoost();
+      const ageHours = '(EXTRACT(EPOCH FROM CURRENT_TIMESTAMP - "createdAt") / 3600)';
+      proj.baseScore = `("baseScore" + (CASE WHEN "tagCommentType" = 'SUBFORUM'
+        THEN GREATEST(
+          ${base},
+          ${magnitude} * (1 - POW(${ageHours} / ${duration}, ${exponent}))
+        )
+        ELSE 0 END)) as "baseScore"`;
+      break;
+  }
+  return Object.values(proj);
+}
+
+const getBatchItemsPg = async <T extends DbObject>(collection: CollectionBase<T>, inactive: boolean) => {
+  const {collectionName} = collection;
+  if (!["Posts", "Comments"].includes(collectionName)) {
+    return [];
+  }
+
+  const db = getSqlClientOrThrow();
+  const singleVotePower = getSingleVotePower();
 
-  const items = await itemsPromise;
-  const itemsArray = await items.toArray();
+  return db.any(`
+    SELECT
+      q.*,
+      ns."newScore",
+      ABS("score" - ns."newScore") > $1 AS "scoreDiffSignificant",
+      (EXTRACT(EPOCH FROM CURRENT_TIMESTAMP - "postedAt") / 3600) > ($2 * 24) AS "oldEnough"
+    FROM (
+      SELECT ${getPgCollectionProjections(collectionName).join(", ")}
+      FROM "${collectionName}"
+      WHERE
+        "postedAt" < CURRENT_TIMESTAMP AND
+        ${inactive ? '"inactive" = TRUE' : '("inactive" = FALSE OR "inactive" IS NULL)'}
+    ) q, LATERAL (SELECT
+      "baseScore" / POW(EXTRACT(EPOCH FROM CURRENT_TIMESTAMP - "postedAt") / 3600 + 2, 1.08) AS "newScore"
+    ) ns
+  `, [singleVotePower, INACTIVITY_THRESHOLD_DAYS]);
+}
+
+const getBatchItems = async <T extends DbObject>(collection: CollectionBase<T>, inactive: boolean) =>
+  Posts.isPostgres()
+    ? getBatchItemsPg(collection, inactive)
+    : getBatchItemsMongo(collection, inactive);
+
+export const batchUpdateScore = async ({collection, inactive = false, forceUpdate = false}) => {
```

Comment:
Once again, let's try to test this (manually and/or automatedly) pretty thoroughly before launch.